### PR TITLE
Logic linking

### PR DESCRIPTION
**Pull request by DavidWoodward42**
I know there are merge conflicts I'm fixing and changing them now

### CODE REVIEW

**Comment by DavidWoodward42** [comment]
Quick question, why was that thing added at the bottom but not used?

File: conf/logback.xml [hunk 1]
```diff
@@ -47,13 +47,13 @@
 
     <logger name="uk.gov" level="INFO"/>
 
-    <logger name="application" level="DEBUG"/>
```
Comment (jameshforster) [inline conf/logback.xml]: Please reset the logging levels.

File: test/connectors/AuthorisationConnectorSpec.scala [hunk 2]
```diff
@@ -64,19 +64,15 @@ class AuthorisationConnectorSpec extends UnitSpec with MockitoSugar with OneAppP
...
       }
 
       "return an set of enrolments" in {
-        result.enrolments shouldBe a[Set[Enrolment]]
-      }
-
-      "return a set of enrolments that contains the key HMRC_AGENT_AGENT_KEY" in {
-        result.enrolments.contains(Enrolment("HMRC_AGENT_AGENT_KEY", Seq(Identifier("Identifier", "Value")), "State")) shouldBe true
+        result.enrolmentsUrl shouldBe "enrolment-uri"
       }
```
Comment (jameshforster) [inline test/connectors/AuthorisationConnectorSpec.scala]: Is this not missing the tests for the getEnrolmentsResponse method?

File: test/controllers/RelationshipControllerSpec.scala [hunk 3]
```diff
@@ -16,90 +16,131 @@
...
+    "the model submitted cannot be parsed into a SubmissionModel" should {
 
-        when(mockHttpDES.POST[JsValue, HttpResponse](ArgumentMatchers.any(), ArgumentMatchers.any(),
-          ArgumentMatchers.any())(ArgumentMatchers.any(), ArgumentMatchers.any(), ArgumentMatchers.any()))
-          .thenReturn(Future.successful(HttpResponse(responseStatus = 500)))
+      val submissionModel = "Not a submission model"
+      val request = FakeRequest().withJsonBody(Json.toJson(submissionModel))
 
-        lazy val result = await(controller.createRelationship(request))
+      "return a status of 404" in {
```
Comment (jameshforster) [inline test/controllers/RelationshipControllerSpec.scala]: Rename test so status matches correct code.

**Pull request merged**